container/controller.go

```diff
@@ -790,103 +790,106 @@ func (c *Controller) doHealthCheck(cancel <-chan struct{}, key health.HealthStat
 	})
 }
 
+func (c *Controller) exportEndpoint(endpoint applicationendpoint.ApplicationEndpoint) {
+	bind := zkservice.ImportBinding{
+		Application:    endpoint.Application,
+		Purpose:        "import", // Punting on control center dynamic imports for now
+		PortNumber:     endpoint.ProxyPort,
+		VirtualAddress: endpoint.VirtualAddress,
+	}
+	exports := make([]registry.ExportDetails, 1)
+	exports[0] = registry.ExportDetails{
+		ExportBinding: zkservice.ExportBinding{
+			Application: endpoint.Application,
+			Protocol:    endpoint.Protocol,
+			PortNumber:  endpoint.ContainerPort,
+		},
+		PrivateIP:  endpoint.ContainerIP,
+		HostIP:     endpoint.HostIP,
+		InstanceID: 0,
+	}
+	c.endpoints.UpdateRemoteExports(bind, exports)
+	return
+}
+
+func (c *Controller) createControlplaneEndpoints() map[string][]applicationendpoint.ApplicationEndpoint {
```

Comment:
I'll move that out to its own spot and just create the registry.ExportDetails that they get turned into, like we just discussed.